### PR TITLE
Compile assets as part of docs generate

### LINKED ISSUE #2072: Compile arbitrary files in `target` directory
### Describe the feature
As a dbt user, I want to be able to include images in my `target` directory, so that my docs can render the images.

Currently, to show images in the docs, you can do something like this:
```
!["Jaffle Shop ERD"](jaffle_shop_erd.png "Jaffle Shop ERD")
```
But this only works if `jaffle_shop_erd.png` is in the `target` directory.

Thoughts:
* Group these assets into a `static` directory to give a user control over what is and isn't compiled.
* Should this have a config in the `dbt_project.yml` like this:
```
asset_paths: ['static']
```
(I have no idea what the right word is here)

### Describe alternatives you've considered
N/A

### Additional context
N/A

### Who will this benefit?
users of the documentation feature

### PR DESCRIPTION
**Pull request by clrcrl**
resolves #2072

### Description

When a user runs `dbt docs generate`, any files in their `assets` directory (configurable!) will be compiled to the `target/assets` directory — useful for adding images to documentation

### Extra validation

I added an image to my own project, at `assets/dbt-logo.png`, and then added it as a description:
```
version: 2

models:
  - name: customers
    description: "![dbt Logo](assets/dbt-logo.png)"
```

It totally works 😎 
Screen Shot 2020-07-21 at 12 53 05 PM

### Checklist
 - [x] I have signed the [CLA](https://docs.getdbt.com/docs/contributor-license-agreements)
 - [x] I have run this code in development and it appears to resolve the stated issue
 - [x] This PR includes tests, or tests are not required/relevant for this PR
 - [x] I have updated the `CHANGELOG.md` and added information about my change to the "dbt next" section.

### CODE REVIEW

**Comment by beckjake** [comment]
>Is having this as part of docs generate a reasonable solution?

Sure, I don't see why not.

>Should I have a test that checks for the presence of a compiled file? If so, can you point me to any other tests that assert that a file exists?

Yeah. Check out the `040_init_test` integration tests.

File: test/unit/test_config.py [hunk 1]
```diff
@@ -580,6 +580,7 @@ def test_defaults(self):
         self.assertEqual(project.test_paths, ['test'])
         self.assertEqual(project.analysis_paths, [])
         self.assertEqual(project.docs_paths, ['models', 'data', 'snapshots', 'macros'])
+        self.assertEqual(project.asset_paths), ['static'])
```
Comment (beckjake) [inline test/unit/test_config.py:583]:
```suggestion
        self.assertEqual(project.asset_paths, ['static'])
```

File: core/dbt/config/project.py [hunk 2]
```diff
@@ -401,6 +402,7 @@ def from_project_config(
         )
 
         docs_paths: List[str] = value_or(cfg.docs_paths, all_source_paths)
+        asset_paths: List[str] = value_or(cfg.asset_paths, ['assets'])
```
Comment (clrcrl) [inline core/dbt/config/project.py:405]: heads up, this differs from the original ticket that had the default as `['static']`, but I prefer the consistency

File: CHANGELOG.md [hunk 3]
```diff
@@ -9,6 +9,7 @@
 - Add more helpful error message for misconfiguration in profiles.yml ([#2569](https://github.com/fishtown-analytics/dbt/issues/2569), [#2627](https://github.com/fishtown-analytics/dbt/pull/2627))
 ### Fixes
 - Adapter plugins can once again override plugins defined in core ([#2548](https://github.com/fishtown-analytics/dbt/issues/2548), [#2590](https://github.com/fishtown-analytics/dbt/pull/2590))
+- Compile assets as part of docs generate ([#2072](https://github.com/fishtown-analytics/dbt/issues/2072), [#2623](https://github.com/fishtown-analytics/dbt/pull/2623/files#))
```
Comment (clrcrl) [inline CHANGELOG.md:12]: I'll fix this lol
```suggestion
- Compile assets as part of docs generate ([#2072](https://github.com/fishtown-analytics/dbt/issues/2072), [#2623](https://github.com/fishtown-analytics/dbt/pull/2623))
```

**Comment by clrcrl** [comment]
Also @drewbanin — can you add your comments re: whether this should work in dbt Cloud?

**Comment by jtcohen6** [comment]
> The only question I have is: should this default to `['assets']`, or to `[]`? Maybe you should have to opt-in to have dbt copy your files around? @jtcohen6 I guess this question is for you

Great point, I think this should be opt-in. The majority of users shouldn't need to worry about an `assets` folder (or even know such a thing exists) until they need it and read the docs about it, at which point adding an extra config line to `dbt_project.yml` feels appropriate. I'd hate to have someone accidentally add an entirely unrelated root-level directory named `assets` and be surprised.

**Comment by clrcrl** [comment]
I was going to argue the counter point — every thing else has a default directory.

But, turns out that `analyses` don't have a [default directory](https://github.com/fishtown-analytics/dbt/blob/dev/marian-anderson/core/dbt/config/project.py#L395)! TIL!

**Comment by clrcrl** [comment]
@drewbanin — assigning you since you have context about the potential security implications of this PR

**Comment by clrcrl** [comment]
Just rebased and pushed :)

**Pull request merged**